src/mds/Server.cc

```diff
@@ -2139,7 +2195,26 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
 
   if (mdr->killed) {
     dout(10) << "request " << *mdr << " was killed" << dendl;
-    return;
+    //if the mdr is a "batch_op" and it has followers, pick a follower as
+    //the new "batch_head" and go on processing the new one.
+    if (mdr->is_batch_op() && mdr->is_batch_head && mdr->batch_reqs.size()) {
+      if (mdr->client_request->get_op() == CEPH_MDS_OP_GETATTR) {
+	auto new_batch_head = *(mdr->batch_reqs.begin());
+	mdr->batch_reqs.erase(new_batch_head);
+	for (auto r : mdr->batch_reqs) {
+	  r->batch_head = new_batch_head;
+	  new_batch_head->batch_reqs.insert(r);
+	}
+	new_batch_head->batch_head.reset();
+	mdr = new_batch_head;
+	CInode* in = mdr->in[0];
+	int mask = mdr->client_request->head.args.getattr.mask;
+	auto fin = in->batch_ops[mask];
+	fin->set_request(new_batch_head);
```

Comment:
not dispatch the new batch head?

Reply:
@ukernel Hi, thanks for the review:-)
The new batch head is set to the current mdr and will be dispatched later in Server::dispatch_client_request method as other mdrs. It seems that I forgot to reset mdr->killed. I'll correct that:-)

Reply:
mds only dispatches head batch request. now the head request is killed,  you need to dispatch the new head request.

Reply:
you are right, but you forget to set mdr->is_batch_head = true;

Reply:
when choosing the new head, you need to make sure it's not killed